app/src/app/api/v0/admin/project/[projectId]/hiap-jobs/route.ts

```diff
@@ -0,0 +1,193 @@
+ *                         enum: [pending, running, completed, failed]
+ *                       createdAt:
+ *                         type: string
+ *                         format: date-time
+ *       400:
+ *         description: Bad Request - Invalid parameters (invalid UUID, year out of range, invalid actionType)
+ *       401:
+ *         description: Unauthorized
+ *       404:
+ *         description: Project not found
+ */
+export const GET = apiHandler(async (req: NextRequest, { params }) => {
+  const { projectId } = params;
+  const { searchParams } = new URL(req.url);
+  const year = searchParams.get("year");
+  const actionType = searchParams.get("actionType");
+
+  // Validate all parameters using Zod
+  const validationResult = hiapJobsQuerySchema.safeParse({
+    projectId,
+    year,
+    actionType,
+  });
+
+  if (!validationResult.success) {
+    const errors = validationResult.error.errors
+      .map((err) => `${err.path.join(".")}: ${err.message}`)
+      .join(", ");
+    throw new createHttpError.BadRequest(`Validation failed: ${errors}`);
+  }
+
+  const {
+    projectId: validatedProjectId,
+    year: validatedYear,
+    actionType: validatedActionType,
+  } = validationResult.data;
+
+  // Verify project exists
+  const project = await db.models.Project.findByPk(validatedProjectId);
+  if (!project) {
+    throw new createHttpError.NotFound("Project not found");
+  }
+
+  // Get all HIAP rankings for cities in this project with required filters
+  const rankings = await db.models.HighImpactActionRanking.findAll({
+    include: [
+      {
+        model: db.models.Inventory,
+        as: "inventory",
+        attributes: ["inventoryId", "year"],
+        where: { year: validatedYear },
+        include: [
+          {
+            model: db.models.City,
+            as: "city",
+            where: {
+              projectId: validatedProjectId,
+            },
+            attributes: ["cityId", "name", "locode"],
+          },
+        ],
+      },
+    ],
+    where: {
+      type: validatedActionType,
+    },
+    attributes: ["id", "jobId", "status", "type", "created"],
+  });
+
+  // Transform the data to match the expected response format
+  const hiapJobs = rankings.map((ranking) => {
+    const rankingWithIncludes = ranking as HighImpactActionRankingWithIncludes;
+    const city = rankingWithIncludes.inventory?.city;
+    const inventory = rankingWithIncludes.inventory;
+
+    if (!city || !inventory) {
+      return null;
+    }
+
+    return {
+      cityId: city.cityId,
+      cityName: city.name,
+      inventoryId: inventory.inventoryId,
+      year: inventory.year,
+      taskId: ranking.jobId,
+      actionType: ranking.type,
+      status: ranking.status,
+      createdAt: ranking.created?.toISOString(),
+    };
+  });
+
+  logger.info(
+    `HIAP Jobs API called for project ${validatedProjectId}, year: ${validatedYear}, actionType: ${validatedActionType}, found ${hiapJobs.length} jobs`,
+  );
```

Comment:
It might make sense to pass this as a data object into the logger as it's useful for structured logging. See here for more details about logging structured data with Pino (which is the tool we're using for logging): https://getpino.io/#/docs/api?id=mergingobject